### ISSUE TITLE
Replace HashFS with new class FileHashStore

### ISSUE THREAD

**Issue by doulikecookiedough** [DataONEorg/hashstore#23]
Our current Python implementation initializes the object and sysmeta directories separately, which results in loading all the methods each time the directories are initialized. Although this is not a critical issue, it is still inefficient. To address this, we should update HashStore to initialize both directories with a single call.

However, this means major changes will have to be made to the `hashfs` class, on top of what is already extended. To improve code readability and avoid confusion, we should organize and consolidate our extended methods into a new class, `HashStoreFS` and only retain the methods utilized from `hashfs`. Additionally, since the `hashfs` library has not been supported for some time, creating our own class based on `hashfs` will mitigate future maintenance overhead and truly keep our codebase in a single repository.

**Comment by doulikecookiedough** [comment]
Refactor methods in `HashStore` and `HashFSExt` into a class `FileHashStore` in preparation for a refactor that implements an abstract base class/subclass `HashStore`.

**Comment by doulikecookiedough** [comment]
After attempting to optimize `Hashstore` by initializing it only once, I found that the added complexity to the existing codebase outweighed the potential benefits (contrary to my expectations). The refactoring process eventually led me down a path where `Hashstore` required the duplication and/or modification of several methods to differentiate between `objects` and `sysmeta` directories. Furthermore, the changes I planned to make appear less robust. I have decided to accept the python implementation and initialization of `Hashstore` as it currently stands and move forward with [refactoring FileHashStore to implement an interface (Hashstore)](https://github.com/DataONEorg/hashstore/issues/26).

I will continue to consider how to optimize `HashStore`'s initialization process as I work through other issues.
